fv3core/decorators.py

```diff
@@ -261,17 +278,23 @@ def decorated(
         domain: Index3D,
         **kwargs,
     ):
-        stencil_config = global_config.get_stencil_config()
-        if isinstance(origin, Mapping):
-            origin_key: Hashable = tuple(sorted(origin.items(), key=lambda x: x[0]))
-            origin_tuple: Tuple[int, ...] = origin["_all_"]
-        else:
+        try:
+            stencil_config = stencil_config_holder[0]
+        except IndexError:
+            stencil_config = global_config.get_stencil_config()
+            stencil_config_holder.append(stencil_config)
```

Comment:
Could `stencil_config_holder` get initialized with `global_config.get_stencil_config()` as its only element?

Reply:
Unfortunately not, the reason it's a holder and not the config itself is that the configuration isn't available until the first call happens.